common/src/main/java/io/netty/util/internal/ClassInitializerUtil.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2021 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.util.internal;
+
+/**
+ * Utility which ensures that classes are loaded by the {@link ClassLoader}.
+ */
+public final class ClassInitializerUtil {
+
+    private ClassInitializerUtil() { }
+
+    /**
+     * Preload the given classes and so ensure the {@link ClassLoader} has these loaded after this method call.
+     *
+     * @param classLoader   the {@link ClassLoader}
+     * @param classNames    the classes to load.
+     */
+    public static void tryLoadClasses(ClassLoader classLoader, String... classNames) {
+        for (String className: classNames) {
+            tryLoadClass(classLoader, className);
+        }
+    }
+
+    /**
+     * Preload the given classes and so ensure the {@link ClassLoader} has these loaded after this method call.
+     *
+     * @param classLoader   the {@link ClassLoader}
+     * @param classes    the classes to load.
+     */
+    public static void tryLoadClasses(ClassLoader classLoader, Class<?>... classes) {
+        for (Class<?> clazz: classes) {
+            tryLoadClass(classLoader, clazz.getName());
+        }
+    }
+
+    private static void tryLoadClass(ClassLoader classLoader, String className) {
+        try {
+            // Load the class and also ensure we init it which means its linked etc.
+            Class.forName(className, true, classLoader);
+        } catch (Exception e) {
```

Comment:
@chrisvest I wonder if we should ignore `ClassNotFoundException` ? Also do we want to do anything special for `SecurityException` ?

Reply:
Because those two would definitely happen later at use sites? Yeah, I suppose we can do that.

Reply:
yep ...

Reply:
@chrisvest done... please have a final look

Reply:
Grrr! checked exceptions!!!    ???

    try {
        ...
    } catch (RuntimeException|Error e) {
         throw e; // let Errors and other runtime exceptions through
    } catch (Exception e) {
        new RuntimeException( ... )
    }

Reply:
@ChrisHegarty already fixed... PTAL again

Reply:
Unsure of this since I don't know how it works in the JDK, but could the deadlock still be possible if `CNFE` (and maybe `SE`) aren't caught here? The class won't be loaded so would it be reattempted in the JNI_OnLoad and as such re-enter the critical sections while searching the classpath? If so might be better to just not exclude them here.

Reply:
I think it should be fine